### ISSUE TITLE
webpack does not update dist/ time stamps, causing unnecessary rebuilds

### ISSUE THREAD

**Issue by ptoscano** [cockpit-project/starter-kit#563]
There is an issue in the dependencies of the current `Makefile`, which triggers a never-ending run of webpack.

In a clean checkout, run the following steps:
```bash
$ make prepare-check 
# if it exists already, npm install won't update it; force that so that we always get up-to-date packages
rm -f package-lock.json
[...]
From https://github.com/cockpit-project/cockpit
 * tag               267        -> FETCH_HEAD
$ make prepare-check 
make: Nothing to be done for 'prepare-check'.
```
Now let's brick it by simply simulating a change in one of the javascript sources:
```
$ touch src/app.jsx 
$ make prepare-check 
NODE_ENV=production node_modules/.bin/webpack
webpack compiled successfully
$ make prepare-check 
NODE_ENV=production node_modules/.bin/webpack
webpack compiled successfully
[... and so on ...]
```

**Comment by martinpitt** [comment]
Confirmed, and this happens with `make` as well, so it's not just a confusion  with different `$NODE_ENV`s. The problem is that running webpack does not actually update the timestamp of dist/manifest.json (our "index" file for the makefile rule).

There's another issue: If you actually switch NODE_ENV, then the old dist/ directory will also not be cleaned, and you get both old and new files. webpack really ought to clean up dist/, and I thought in cockpit it actually does that. I'll see if I find a webpack option to do that, otherwise let's just `rm` it before.

**Comment by martinpitt** [comment]
Fixed both issues in PR #564
